### PR TITLE
Allow for external mods to add their own books.

### LINKED ISSUE #5: [request] Multiple books
Would it be possible to add multiple books?

### LINKED ISSUE #7: Example tab always present without user intervention
I know the example tab is intended to help users of the mod (e.g. modpack makers) understand the config file.

My issue is that when using this mod as a dependency to add my mod's guide book, the example tab will always be there unless the user edits their config file to be something like: 

```
manual {

    exampletab {
    }

}
```

So off the top of my head this leaves a couple of options
- Only add the example tab if no other tabs have been added via api before `postInit`
    - Perhaps also having a config option to force example creation?
- Shade the necessary classes into my mod(s)

I prefer the first option because I would like to use it in at least a couple mods (that don't depend on each other).

What say ye?

### PR DESCRIPTION
**Pull request by thiakil**
I got bored and had a crack at making this work. I am aware you have a branch for it, but a ForgeRegistry seemed a bit overkill to me. I won't be offended if you still wanna do it your way.

Seems to work fine for my mod's book, and makes #7 a non-issue

Closes #5 and closes #7.

### CODE REVIEW

**Comment by Vexatos** [comment]
I don't like not having some kind of registry. The way you have it, there is no way for any mod to get another mod's manual to add tabs to it.

**Comment by thiakil** [comment]
Why would a mod *want* someone else to add pages to their manual unless they specifically allow it?

**Comment by Vexatos** [comment]
In case you write an addon for a mod that did not intend to have addons? I myself have written at least three of those.

**Comment by thiakil** [comment]
I'd argue you should be having your own book if the parent mod did not provide a way to access it (or like, you *ask* them)

**Comment by Vexatos** [comment]
>or like, you ask them

...and the mod hadn't had a new release in half a year, and isn't actively developed anymore.

There is no need to argue about this, there _are_ cases where a registry would be useful.

**Comment by thiakil** [comment]
So you add your own book, or go the difficult way if you insist on forcibly inserting your stuff.

> there are cases where a registry would be useful

Would you care to enlighten me?

File: src/main/java/li/cil/manual/api/detail/ManualDefinition.java [hunk 1]
```diff
@@ -138,4 +150,20 @@
      * @param path the path to navigate to.
      */
     void navigate(String path);
+
+    void setDefaultPage(final String defaultPage);
```
Comment (Vexatos) [inline src/main/java/li/cil/manual/api/detail/ManualDefinition.java]: Shouldn't these be documented as well?

Comment (thiakil) [reply src/main/java/li/cil/manual/api/detail/ManualDefinition.java]: yes, good point. IIRC they were the functions that used to be static on the `Impl` class

**Comment by thiakil** [comment]
As an aside, I think the book item should use `onItemUseFirst` instead of `onItemUse`, as otherwise you cant right click lookup gui-possessing blocks

**Comment by Vexatos** [comment]
Can't you still open it by shift-right clicking?

**Comment by thiakil** [comment]
Sure, but seems annoying when everything else does it with just a click

**Comment by talonos2** [comment]
Thanks from me as well! This functionality excites me.

Will these changes make it to the 1.12 branch any time in the near future? If not, I understand; it seems like this mod isn't being actively developed anymore, and nobody has a right to the authors time. But I'm interested if this enhancement will be available for 1.12 or if I should continue seeking an alternate solution.

**Comment by Vexatos** [comment]
The mod is being developed, just slowly. A 1.12 release is already out.

**Comment by talonos2** [comment]
My bad. Shoulda hit "Refresh" on the 1.12 repo I had open before posting a comment.

Thank you both for your work!

**Pull request merged**